src/devices/bus/nes/nes_pcb.hxx

```diff
@@ -661,6 +659,51 @@ void nes_cart_slot_device::call_load_pcb()
 			m_cart->set_n163_vol(n163_get_submapper_num(get_feature("n163-vol")));
 	}
 
+	if (m_pcb_id == BMC_EL860947C)
+	{
+		const char *size = get_feature("outer-prg-size");
+		if (size != nullptr)
+		{
+			if (!strcmp(size, "128K"))
+				m_cart->set_outer_prg_size(128);
+			else if (!strcmp(size, "256K"))
+				m_cart->set_outer_prg_size(256);
+			else
+				fatalerror("Unexpected outer-prg-size: %s\n", size);
+		}
+		else
+			fatalerror("EL86XC must define outer-prg-size!\n");
```

Comment:
Wouldn’t you be better of doing `scanf("%u%c", ...)` or equivalent, and then checking that it’s a power-of-two (`!(x & (x -1))`)?  Is there a good reason not to use a default of 128k or something?  Do you really need to throw fatal errors here, potentially bringing down the program, rather than just causing the media load to fail?

Same comments apply for all these properties.

Reply:
In addition to integrating your suggestions I reorganized it to not read in these features per board but instead read per feature, i.e. there's one block to get outer PRG size and one to get outer CHR size for all boards. I don't anticipate there are many more NES boards that will use these features, so hopefully this new version suffices in general.